OuterWildsRandomSpeedrun/SpeedrunHarmonyPatches.cs

```diff
@@ -52,4 +53,23 @@ public static void DebugInputManager_Start_Postfix()
   {
     SpeedrunState.IsGameStarted = true;
   }
+
+  // Disable achievements while the mod is running, since many warps can accidentally
+  // trigger ones like Hotshot.
+  [HarmonyPostfix]
+  [HarmonyPatch(typeof(Achievements), nameof(Achievements.Earn))]
+  public static bool Achievements_Earn_Postfix(Achievements.Type type)
+  {
+    var modHelper = OuterWildsRandomSpeedrun.OuterWildsRandomSpeedrun.Instance.ModHelper;
+    if (SpeedrunState.ModEnabled)
+    {
+      modHelper.Console.WriteLine($"Skipping achievement {type} during Nomai Grand Prix run.", MessageType.Info);
+      return false;
+    }
+    else
+    {
+      modHelper.Console.WriteLine($"Allowing player to earn achievement {type} while Nomai Grand Prix is inactive.", MessageType.Info);
```

Comment:
I would suggest not logging this so we are impacting the base game as little as possible. (If the mod's disabled, it shouldn't be doing anything)

Reply:
Also I think this code is still going to short-circuit the awarding of the achievement since you're using `return` here.

Reply:
If the player's enabled the mod in the mod manager, we're always going to log something when the mod is first initialized. Logging seems harmless and won't be noticed by most players. If it bothers them, they can always disable the mod in the mod manager.

Reply:
I'm more inclined to @cmaines perspective on this. The reason we're hiding the mod behind a menu item is so that they don't have to turn it on and off. I agree that logging has no visible impact so I don't feel strongly about it, but I think that the argument that they can turn the mod off doesn't hold water to me.

Reply:
To clarify, I think every user-visible change in the game itself should be hidden when the player doesn't select Nomai Grand Prix. What makes me want this log line anyway is that it _isn't_ visible for the majority of players; I seriously doubt most (any?) users will both check the logs _and_ be annoyed by the presence of this low-frequency item. There's a tangible benefit to us to keep the logging there, both for debugging during our own development and if a user reports a bug. My point is just that for any users that both check the logs when our mod is enabled and don't want to see our log line, they still have access to the break-glass "turn the mod off" option.